src/DssProxyRegistry.sol

```diff
@@ -31,23 +32,23 @@ contract DssProxyRegistry {
     }
 
     function build(address owner_) external returns (address payable proxy) {
-        require(proxies[owner_] == address(0), "DssProxyRegistry/proxy-already-registered-to-owner"); // Not allow new proxy if the user already has one and remains being the owner
+        proxy = payable(proxies[owner_]);
+        require(proxy == address(0) || DssProxy(payable(proxy)).owner() != owner_, "DssProxyRegistry/proxy-registered-to-owner"); // Not allow new proxy if the user already has one and remains being the owner
         seed[owner_]++;
         uint256 salt = _salt(owner_);
         bytes memory code = _code(owner_);
         assembly {
             proxy := create2(0, add(code, 0x20), mload(code), salt)
         }
         proxies[owner_] = proxy;
+        isProxy[proxy] = 1;
     }
 
-    function claim(address payable proxy, address src, address dst) external {
-        address currentDst = proxies[dst];
-        require(currentDst == address(0) || DssProxy(payable(currentDst)).owner() != dst, "DssProxyRegistry/proxy-registered-to-owner"); // Not allow new proxy if the user already has one and remains being the owner
-        require(DssProxy(proxy).owner() == dst, "DssProxyRegistry/proxy-not-owned-by-dst");
-        require(proxies[src] == proxy, "DssProxyRegistry/must-provide-a-valid-previous-owner");
-
-        proxies[src] = address(0);
-        proxies[dst] = proxy;
+    function claim(address payable proxy) external {
+        require(isProxy[proxy] == 1, "DssProxyRegistry/not-proxy-from-this-registry");
+        address owner = DssProxy(proxy).owner();
+        address payable prevProxy = payable(proxies[owner]);
+        require(prevProxy == address(0) || DssProxy(prevProxy).owner() != owner, "DssProxyRegistry/proxy-registered-to-owner"); // Not allow new proxy if the user already has one and remains being the owner
+        proxies[owner] = proxy;
```

Comment:
My only issue is that the prevProxy is going to be orphaned after this gets claimed.

could we add something like:

```suggestion
        address payable prevProxy = payable(proxies[owner]);
        require(prevProxy == address(0) || DssProxy(prevProxy).owner() != owner, "DssProxyRegistry/proxy-registered-to-owner"); // Not allow new proxy if the user already has one and remains being the owner
        proxies[DssProxy(prevProxy).owner()] = prevProxy;
        proxies[owner] = proxy;
```

Reply:
Well the problem with that is the new owner of your `prevProxy` might have already another proxy.
I feel it is just better to assign the main proxy and if there is an orphan proxy resulting from this first action, then it can be registered back in the proper `claim` to it.

Reply:
Yeah i could see that.  then of course we are not enforcing one proxy to one address.  Not sure we need that to be iron clad since it would require calling from the proxy to the registry on ownership change or handling the ownership change through the registry only.

as long as we are okay not having strong enforcement here, i think we should leave this as is and accept that some proxies may not appear in the registry

Reply:
Yeah I feel that's the correct balance to improve the actual functionality without designing a much more complex solution.